internal/controllers/utils/utils.go

```diff
@@ -163,8 +166,90 @@ func GetDeploymentVolumeMounts(serverName string) []corev1.VolumeMount {
+	// that will eventually execute the server. Note that the file may not exist,
+	// but we can't check it here as that will be a different pod.
+	if backendToken != "" {
+		return fmt.Sprintf("--backend-token=%s", backendToken)
+	}
+
+	return fmt.Sprintf("--backend-token-file=%s", defaultBackendTokenFile)
+}
+
+// getACMNamespace will determine the ACM namespace from the multiclusterengine object.
+//
+// multiclusterengine object sample:
+//
+//	apiVersion: multicluster.openshift.io/v1
+//	kind: MultiClusterEngine
+//	metadata:
+//	  labels:
+//	    installer.name: multiclusterhub
+//	    installer.namespace: open-cluster-management
+func getACMNamespace(ctx context.Context, c client.Client) (string, error) {
+	// Get the multiclusterengine object.
+	multiClusterEngine := &unstructured.Unstructured{}
+	multiClusterEngine.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "multicluster.openshift.io",
+		Kind:    "MultiClusterEngine",
+		Version: "v1",
+	})
+	err := c.Get(ctx, client.ObjectKey{
+		Name: "multiclusterengine",
+	}, multiClusterEngine)
+
+	if err != nil {
+		oranUtilsLog.Info("[getACMNamespace] multiclusterengine object not found")
+		return "", fmt.Errorf("multiclusterengine object not found")
+	}
+
+	// Get the ACM namespace by looking at the installer.namespace label.
+	multiClusterEngineMetadata := multiClusterEngine.Object["metadata"].(map[string]interface{})
+	multiClusterEngineLabels, labelsOk := multiClusterEngineMetadata["labels"]
+
+	if labelsOk {
+		acmNamespace, acmNamespaceOk := multiClusterEngineLabels.(map[string]interface{})["installer.namespace"]
+
+		if !acmNamespaceOk {
+			return "", fmt.Errorf("multiclusterengine labels do not contain the installer.namespace key")
+		}
+		return acmNamespace.(string), nil
+	}
+
+	return "", fmt.Errorf("multiclusterengine object does not have expected labels")
+}
+
+// getSearchAPI will dynamically obtain the search API.
+func getSearchAPI(ctx context.Context, c client.Client, orano2ims *oranv1alpha1.ORANO2IMS) (string, error) {
+	// Find the ACM namespace.
+	acmNamespace, err := getACMNamespace(ctx, c)
+	if err != nil {
+		return "", err
+	}
+
+	// Split the Ingress to obtain the domain for the Search API.
+	// searchAPIBackendURL example: https://search-api-open-cluster-management.apps.lab.karmalabs.corp
+	// IngressHost example:         o2ims.apps.lab.karmalabs.corp
+	// Note: The domain could also be obtained from the spec.host of the search-api route in the
+	// ACM namespace.
+	ingressSplit := strings.Split(orano2ims.Spec.IngressHost, ".apps")
+	if len(ingressSplit) != 2 {
+		return "", fmt.Errorf("the searchAPIBackendURL could not be obtained from the IngressHost. " +
+			"Directly specify the searchAPIBackendURL in the ORANO2IMS CR or update the IngressHost")
+	}
+	domain := ".apps" + ingressSplit[len(ingressSplit)-1]
+
+	// The searchAPI is obtained from the "search-api" string and the ACM namespace.
+	searchAPI := "https://" + "search-api-" + acmNamespace + domain
```

Comment:
For a different pull request, but note that as we will be running inside the same cluster than the search API we can use directly the `search-search-api` service instead of creating a new route that points to it. So the URL could be:

```
https://search-search-api.open-cluster-management.svc.cluster.local:4010
```

That can simplify things, and reduces the number of hops to get to the backend.